pdns/recursordist/nod.cc

```diff
@@ -40,46 +40,46 @@ namespace filesystem = boost::filesystem;
 
 std::mutex PersistentSBF::d_cachedir_mutex;
 
-void PersistentSBF::remove_tmp_files(const filesystem::path& p, std::lock_guard<std::mutex>& /* lock */)
+void PersistentSBF::remove_tmp_files(const filesystem::path& path, std::lock_guard<std::mutex>& /* lock */)
 {
   Regex file_regex(d_prefix + ".*\\." + bf_suffix + "\\..{8}$");
-  for (filesystem::directory_iterator i(p); i != filesystem::directory_iterator(); ++i) {
-    if (filesystem::is_regular_file(i->path()) && file_regex.match(i->path().filename().string())) {
-      filesystem::remove(*i);
+  for (const auto& file : filesystem::directory_iterator(path)) {
+    if (filesystem::is_regular_file(file.path()) && file_regex.match(file.path().filename().string())) {
+      filesystem::remove(file);
     }
   }
 }
 
-// This looks for an old (per-thread) snapshot. The first one it finds,
-// it restores from that. Then immediately snapshots with the current thread id,// before removing the old snapshot
-// In this way, we can have per-thread SBFs, but still snapshot and restore.
-// The mutex has to be static because we can't have multiple (i.e. per-thread)
-// instances iterating and writing to the cache dir at the same time
+// This looks for the newest (per-thread) snapshot it can find and it restores from that. Then
+// immediately snapshots with the current thread id, before removing the old snapshot.
+// In this way, we can have per-thread SBFs, but still snapshot and restore.  The mutex has to be
+// static because we can't have multiple (i.e. per-thread) instances iterating and writing to the
+// cache dir at the same time
 bool PersistentSBF::init(bool ignore_pid)
 {
   auto log = g_slog->withName("nod");
   std::lock_guard<std::mutex> lock(d_cachedir_mutex);
```

Comment:
Unrelated to this PR but it's unclear to me what `d_cachedir_mutex` protects. If it's there to prevent two threads from touching the cache directory at the same time, shouldn't `snapshotCurrent()` hold the mutex?

Reply:
A added a comment to explain this.